omp/components/atomic.hpp

```diff
@@ -85,6 +85,115 @@ void atomic_add(half& out, half val)
 }
 
 
+// There is an error in Clang 17 which prevents us from merging the
+// implementation of double and float. The compiler will throw an error if the
+// templated version is implemented. GCC doesn't throw an error.
+template <typename T>
+struct atomic_store_helper;
```

Comment:
Do we actually need this? Or can we get the same result with just overloaded functions? We only use atomics for a fixed set of types (value types and index types)

Reply:
You're right. I first wanted to use the templated implementation, but as that doesn't work with clang, we can just use simple overloads.